frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx

```diff
@@ -216,8 +225,127 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
         }
     }, [availableClusters, providerSecretMap, fetchSecrets])
 
+    useEffect(() => {
+        const name = 'submariner-broker'
+        const namespace = clusterSet?.metadata?.annotations?.['cluster.open-cluster-management.io/submariner-broker-ns']
+        if (namespace) {
+            getBroker({ name, namespace }).promise.then(broker=>{
+                setGlobalnetEnabled(broker?.spec?.globalnetEnabled ?? false)
+                setGlobalnetDisabled(true)
+            })
+            .catch(() => {
+                setGlobalnetDisabled(false)
+            })
+        } else {
+            setGlobalnetHidden(true)
```

Comment:
@jeswanke so if for some reason Submariner has not created the broker namespace, we hide the globalnet settings? Do we avoid trying to create the Broker in this case?

Reply:
@KevinFCormier if the namespace doesn't exist and they select Globalnet it will definitely fail--so I didn't show it---optionally we can show it and they see an error message

Reply:
Are you saying that if it is hidden, we won't try to create the Broker? @sridhargaddam thoughts?

Reply:
@KevinFCormier if the clusterset doesn't have this annotation, we don't know where to create the broker:

clusterSet?.metadata?.annotations?.['cluster.open-cluster-management.io/submariner-broker-ns']

which is a messed up situation, just trying to fail gracefully

Reply:
The broker-namespace is very fundamental to Submariner as its used to store a lot of objects, so you can expect that it will be present (unless there is some bug in which case Submariner functionality will be broken anyway).

Reply:
In case you think its easy to show some warning when the annotation (cluster.open-cluster-management.io/submariner-broker-ns) is missing on the ManagedClusterSet object, it will be nice. But I don't know the effort involved, so I'll leave it to you do decide.

Reply:
so it never should happen and if it does you should be looking for the emergency exits